bindings/java/src/main/java/org/apache/opendal/layer/RetryNativeLayer.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.opendal.layer;
+
+import java.time.Duration;
+import lombok.Builder;
+import org.apache.opendal.NativeLayer;
+
+@Builder
+public class RetryNativeLayer extends NativeLayer {
+
+    private final boolean jitter;
```

Comment:
Is it a good idea to store a `nativeHandle` (the `RetryLayer` in rust core) in `RetryNativeLayer` class and forward all functions to it? So the `doLayer` can be `doLayer(long nativeHandle, long layerHandle)`.

Besides, I have two naming things:

- How renaming `nativeHandle` to `nativeOperator`?
- Do we need to keep `Native` in `RetryNativeLayer`? How about just `RetryLayer`? I feel like users don't need to know about this concept.

Reply:
> Is it a good idea to store a nativeHandle

No. This causes `RetryNativeLayer` to hold a native handle to release. I'd prefer to avoid this pattern as much as possible. Otherwise, Java developers should always remember to release the native handle, which is quite unfamiliar with the normal programming paradigm.

Reply:
> How about renaming nativeHandle to nativeOperator across the whole package?

It's possible. Let's do it in a follow-up PR. It changes only parameter name so it won't break API also.

Reply:
> Do we need to keep Native in RetryNativeLayer? How about just RetryLayer? I feel like users don't need to know about this concept.

Let me think of it. This can be reasonable.

Reply:
> Let me think of it. This can be reasonable.

Let's keep it as is for now. Still I'm thinking of a layer concept to intercept in Java level, so keep NativeLayers named as NativeLayer leave the room for such possibility.

Reply:
> This causes `RetryNativeLayer` to hold a native handle to release.

Can we implement something like `Drop` for `RetryNativeLayer`? I'm guessing java can do something while the class is not referred. By the way, the memory of `layerHandle` should be freed after `doLayer` has been called.

> Still I'm thinking of a layer concept to intercept in Java level

It's ok to keep to have different layer types.

But users don't need to know about them.

For example, users could call:

```java
op.layer(RetryLayer).layer(TracingLayer).layer(MyLayer)
```

without knowing it's `native` or not.

Reply:
> users could call

I have another round for this and agree with it. Let me modify the interface.

Reply:
> Can we implement something like Drop for RetryNativeLayer? I'm guessing java can do something while the class is not referred. By the way, the memory of layerHandle should be freed after doLayer has been called.

No. This is what try-with-resource and AutoCloseable does, as I implement it in `NativeObject`. This is not the way to go especially you refer to "the memory of layerHandle should be freed after doLayer has been called". It's vague and unclear to handle the ownership and lifetime properly, so I tend to use a POJO as we refactor Metadata as a POJO rather than a NativeObject.

> I'm guessing java can do something

Previously, it's "finalize", while the method is deprecated to be removed.